package o2ovender;

option optimize_for = CODE_SIZE;

// 激活码请求
message get_identifying_code_req
{
    required string uid = 1;  // 手机号码
};

// 注册请求
message register_req
{
    required string uid = 1;               // 手机号码
    required string pass_word = 2;         // 密码
    optional string identifying_code = 3;  // 验证码
};

// 注册响应
message register_resp
{
    required int32 result = 1;  // 登陆结果: 0成功,其它失败
    optional string msg = 2;    // 失败原因
enum ERRCODE
{
    NO_ERR = 0;          // 没有错误
    SERVER_BUSY = 1;     // 服务器繁忙
    ACCOUNT_EMPTY = 2;   // 输入的账号为空
    ACCOUNT_EXISTS = 3;  // 账号已存在
    PASSWD_EMPTY = 4;    // 输入的密码为空
    IDCODE_EMPTY = 5;    // 输入的验证码为空
    IDCODE_WRONG = 6;    // 验证码不正确
 }
    optional ERRCODE errcode = 3 [default = NO_ERR];  // 失败原因代码
};

// 登陆请求
message login_req
{
    required string uid = 1;        // 手机号码
    required string pass_word = 2;  // 密码
};

// 登陆响应
message login_resp
{
    required int32 result = 1;  // 登陆结果: 0成功,其它失败
    optional string msg = 2;    // 失败原因
enum ERRCODE
{
    NO_ERR = 0;              // 没有错误
    SERVER_BUSY = 1;         // 服务器繁忙
    ACCOUNT_EMPTY = 2;       // 输入的账号为空
    ACCOUNT_NOT_EXISTS = 3;  // 账号不存在
    PASSWD_EMPTY = 4;        // 输入的密码为空
    PASSWD_WRONG = 5;        // 密码不正确
 }
    optional ERRCODE errcode = 3 [default = NO_ERR];  // 失败原因代码
};

// 商品列表请求
message commodity_list_req
{
};

message commodity_info
{
    required string id = 1;       //  商品唯一标记
    optional int32 type = 2;      //  商品类型
    optional int32 code = 3;      //  商品编码
    optional string name = 4;     //  商品名称
    optional string img = 5;      //  商品图片
    optional float price = 6;     //  商品价格
    optional float discount = 7;  //  商品折扣价
};

// 商品列表响应
message commodity_list_resp
{
enum ERRCODE
{
    NO_ERR = 0;              // 没有错误
    SERVER_BUSY = 1;         // 服务器繁忙
    NO_LOGIN = 2;            // 没有登录
 }
    required int32 result = 1;  // 结果: 0成功,其它失败
    optional string msg = 2;    // 失败原因
    optional ERRCODE errcode = 3 [default = NO_ERR];  // 失败原因代码
    repeated commodity_info commodity_info = 4;  // 商品列表
};

// 请求
message request
{
enum TYPE
{
    NO_TYPE = 0;
    GET_IDENTIFYING_CODE = 1;
    REGISTER= 2;
    LOGIN = 3;
    COMMODITY_LIST = 4;
}
    required TYPE type = 1 [default = NO_TYPE];
    optional get_identifying_code_req get_identifying_code_req = 2;
    optional register_req register_req = 3;
    optional login_req login_req = 4;
    optional commodity_list_req commodity_list_req = 5;
};

// 响应
message response
{
enum TYPE
{
    NO_TYPE = 0;
    REGISTER= 1;
    LOGIN = 2;
    COMMODITY_LIST = 3;
}
    required TYPE type = 1 [default = NO_TYPE];
    optional login_resp login_resp = 2;
    optional register_resp register_resp = 3;
    optional commodity_list_resp commodity_list_resp = 4;
};
